models/reporter.go

```diff
@@ -622,26 +474,46 @@ func (r *Reporter) SelectFilters(ctx context.Context, periodStartFromSec, period
 	return &result, nil
 }
 
-func (r *Reporter) queryFilters(ctx context.Context, periodStartFromSec, periodStartToSec int64, mainMetricName, query string) ([]*customLabel, float32, error) {
+func (r *Reporter) queryFilters(ctx context.Context, periodStartFromSec,
+	periodStartToSec int64, dimensionName, mainMetricName string, tmplQueryFilter *template.Template, queryDimensions, queryLabels map[string][]string) ([]*customLabel, float32, error) {
 	durationSec := periodStartToSec - periodStartFromSec
 	var labels []*customLabel
-	rows, err := r.db.QueryContext(ctx, fmt.Sprintf(query, mainMetricName), periodStartFromSec, periodStartToSec)
+
+	tmplArgs := struct {
+		MainMetric    string
+		DimensionName string
+		Dimensions    map[string][]string
+		Labels        map[string][]string
+	}{
+		mainMetricName,
+		dimensionName,
+		queryDimensions,
+		queryLabels,
+	}
+
+	var queryBuffer bytes.Buffer
+
+	if err := tmplQueryFilter.Execute(&queryBuffer, tmplArgs); err != nil {
+		return nil, 0, errors.Wrap(err, "cannot execute tmplQueryFilter"+queryBuffer.String())
```

Comment:
`errors.Wrapf(err, "cannot execute tmplQueryFilter %s", queryBuffer)`

The same in other places.